samples/jpegparsetest.cpp

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Sample program to parse jpeg files and obtain its specific metadata
+
+#include <exiv2/exiv2.hpp>
+#include <exiv2/image_types.hpp>
+#include <iostream>
+
+int main(int argc, char* const argv[]) {
+  try {
+    if (argc != 2) {
+      std::cout << "Usage: " << argv[0] << " file\n";
+      return EXIT_FAILURE;
+    }
+
+    auto image = Exiv2::ImageFactory::open(argv[1]);
+    if (image->imageType() == Exiv2::ImageType::jpeg) {
+      Exiv2::JpegImage* jpegImage = dynamic_cast<Exiv2::JpegImage*>(image.get());
```

Comment:
auto jpegImage.

Reply:
makes sense, implicit from the dynamic_cast

Reply:
yep. clang-tidy warns on not using auto.